subprojects/docs/src/samples/userguide/tasks/incrementalTask/groovy/build.gradle

```diff
@@ -41,44 +41,38 @@ task incrementalReverse(type: IncrementalReverseTask) {
 // end::reverse[]
 
 // tag::incremental-task[]
-class IncrementalReverseTask extends DefaultTask {
+abstract class IncrementalReverseTask extends DefaultTask {
+    @Incremental
+    @PathSensitive(PathSensitivity.NAME_ONLY)
     @InputDirectory
-    def File inputDir
+    abstract DirectoryProperty getInputDir()
 
     @OutputDirectory
-    def File outputDir
+    abstract DirectoryProperty getOutputDir()
 
     @Input
-    def inputProperty
+    abstract Property<String> getInputProperty()
 
     @TaskAction
-    void execute(IncrementalTaskInputs inputs) {
-        println inputs.incremental ? 'CHANGED inputs considered out of date'
-                                   : 'ALL inputs considered out of date'
-        // tag::handle-non-incremental-inputs[]
-        if (!inputs.incremental)
-            project.delete(outputDir.listFiles())
-        // end::handle-non-incremental-inputs[]
+    void execute(InputChanges inputChanges) {
+        println(inputChanges.incremental
+            ? 'Executing incrementally'
+            : 'Executing non-incrementally'
+        )
 
-        // tag::out-of-date-inputs[]
-        inputs.outOfDate { change ->
-            if (change.file.directory) return
+        // tag::process-file-changes[]
+        inputChanges.getFileChanges(inputDir).each { change ->
+            if (change.fileType == FileType.DIRECTORY) return
 
-            println "out of date: ${change.file.name}"
-            def targetFile = new File(outputDir, change.file.name)
-            targetFile.text = change.file.text.reverse()
+            println "${change.changeType}: ${change.normalizedPath}"
+            def targetFile = outputDir.file(change.normalizedPath).get().asFile
+            if (change.changeType == ChangeType.REMOVED) {
+                targetFile.delete()
+            } else {
+                targetFile.text = change.file.text.reverse()
```

Comment:
@wolfs How does this ensure that the target directory for the file exists? Is it the `setText()` method that ensures the directory exists? Or is Gradle doing it somehow?

Reply:
The task uses only the name of the input file. So the `targetFile` is a direct child of the output directory. The output directory is created automatically by Gradle.

Reply:
Oh right, I see. Missed that, sorry. I only ask because I want to add an introductory paragraph for the example that explains what it does and what the reader should look out for.